libraries/Microsoft.Bot.Builder.Ai.LUIS/Generator/InstanceData.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Ai.LUIS
+{
+    /// <summary>
+    /// Strongly typed information corresponding to LUIS $instance value.
+    /// </summary>
+    /// <remarks>
+    /// This is a partial class in order to support adding custom meta-data.
+    /// </remarks>
+    public class InstanceData
+    {
+        /// <summary>
+        /// 0-based index in the analyzed text for where entity starts.
+        /// </summary>
+        [JsonProperty("startIndex")]
+        public int StartIndex;
+
+        /// <summary>
+        /// 0-based index of the last character for recognized entity.
+        /// </summary>
+        [JsonProperty("endIndex")]
+        public int EndIndex;
```

Comment:
BUG: The "EndIndex", as defined, is the wrong semantic here. Starting from C, all languages in that family use "1 past the end" semantics. This is a data loss problem waiting to happen.

StartIndex, Length are probably the right semantics. 

C# would normally consume this via:
[String.Substring Subtring](https://msdn.microsoft.com/en-us/library/aka44szs(v=vs.110).aspx), which is defined as:
```
public string Substring(
	int startIndex,
	int length
)
```

Javascript, amusingly, as a 3rd semantic (via [Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring))
`str.substring(indexStart[, indexEnd])`

Where IndexEnd is the EndIndex+1 from what you define in your code above:

From the doc:
>  The index of the **first character to exclude** from the returned substring.

Python uses a Slicer, which is very similar to the Javascript semantic, which is again, different from yours. 
> MyString[a:b] gives you a substring from index a to (b - 1)

Reply:
Fixed